src/peer_digest.cc

```diff
@@ -538,7 +538,14 @@ peerDigestFetchReply(void *data, char *buf, ssize_t size)
 
             assert(fetch->old_entry->mem_obj->request);
 
-            Store::Root().updateOnNotModified(fetch->old_entry, *fetch->entry);
+            try {
+                Store::Root().updateOnNotModified(fetch->old_entry, *fetch->entry);
+            } catch (...) {
+                static const char *reason = "error when updating entry";
+                debugs(20, DBG_IMPORTANT, reason << *fetch->old_entry << ": " << CurrentException);
+                peerDigestFetchAbort(fetch, buf, reason);
```

Comment:
We should not add low-level StoreEntry dump to level-1 messages. Also, we should use a well-known `ERROR:` prefix for cache.log-reported errors. And there is probably no reason to remember the address of that string literal. The compiler knows what it is.

```suggestion
                auto error = "cannot update cached entry";
                debugs(20, DBG_IMPORTANT, "ERROR: " << error << ": " << CurrentException);
                peerDigestFetchAbort(fetch, buf, error);
```

I would also simplify code and code searches at the expense of some minor code duplication:

```suggestion
                debugs(20, DBG_IMPORTANT, "ERROR: Cannot update cached entry: " << CurrentException);
                peerDigestFetchAbort(fetch, buf, "header update failure after a 304 response");
```